nym-vpn-lib/src/mixnet_connect.rs

```diff
@@ -123,9 +123,15 @@ async fn wait_for_connect_response(
 
                         // Handle if the response is from an IPR running an older or newer version
                         if let Some(version) = msg.message.first() {
-                            if *version != nym_ip_packet_requests::CURRENT_VERSION {
-                                log::error!("Received packet with invalid version: v{version}, is your client up to date?");
-                                return Err(Error::InvalidVersion {
+                            if *version > nym_ip_packet_requests::CURRENT_VERSION {
+                                log::error!("Received packet with newer version: v{version}, is your client up to date?");
+                                return Err(Error::ReceivedResponseWithNewVersion {
+                                    expected: nym_ip_packet_requests::CURRENT_VERSION,
+                                    received: *version,
+                                });
+                            } else {
```

Comment:
Wouldn't the correct matching version also go on this branch?

Reply:
yes of course, good spot !